provider/provider.go

```diff
@@ -47,3 +47,10 @@ type Query interface {
 	// Query queries the provider with the given model name.
 	Query(ctx context.Context, modelIdentifier string, promptText string) (response string, err error)
 }
+
+// Background is a provider with background services.
+type Background interface {
```

Comment:
Will rename this to "Service" sure. it runs in the background, but what we basically want to know is if this provider requires a service or not.